kaniko/tests/run.yaml

```diff
@@ -0,0 +1,22 @@
+---
+apiVersion: tekton.dev/v1alpha1
+kind: TaskRun
+metadata:
+  name: kaniko-run
+spec:
+  serviceAccount: %SERVICE_ACCOUNT%
```

Comment:
yes you are right, see my previous comment!

on a default k8 cluster you don't need a service account because by default you don't have any RBAC configured (i may be wrong of the reason tho, but it just works on k8s). On OpenShift (and RBAC enabled k8 clusters) you don't have this by default  and you need to specify a specific SA account for those.